packages/dev/query/src/cli.js

```diff
@@ -130,23 +131,105 @@ function getAsset(v: string) {
   }
 }
 
-function findAsset(v: string) {
+function _findAssetNode(v: string) {
   let assetRegex = new RegExp(v);
   for (let node of assetGraph.nodes.values()) {
     if (
       node.type === 'asset' &&
       assetRegex.test(fromProjectPathRelative(node.value.filePath))
     ) {
-      try {
-        console.log(
-          `${bundleGraph.getAssetPublicId(
-            bundleGraph.getAssetById(node.id),
-          )} ${fromProjectPathRelative(node.value.filePath)}`,
-        );
-      } catch (e) {
-        console.log(fromProjectPathRelative(node.value.filePath));
+      return node;
+    }
+  }
+}
+
+function findAsset(v: string) {
+  let node = _findAssetNode(v);
+  if (node) {
+    try {
+      console.log(
+        `${bundleGraph.getAssetPublicId(
+          bundleGraph.getAssetById(node.id),
+        )} ${fromProjectPathRelative(node.value.filePath)}`,
+      );
+    } catch (e) {
+      console.log(fromProjectPathRelative(node.value.filePath));
+    }
+  }
+}
+
+function findAssetWithSymbol(local: string) {
+  let [, assetId, binding, ref] = nullthrows(
+    local.match(/^\$([^$]+)\$([^$]+)\$(.*)$/),
+    `symbol ${local} could not be resolved`,
+  );
+
+  let asset = bundleGraph._graph.getNodeByContentKey(assetId);
```

Comment:
If you have plugins running after the JS transformer, then `assetId` will be the asset id at the point where the JS transformer ran, and not the final one in the graph. So ideally this would be something like `assetGraph.nodes.find(n => n.type === "asset" && n.value.meta.id === assetId)`

Reply:
Ah interesting! I thought I had found a shortcut 😆